libs/feature/record/src/lib/links/link-utils.ts

```diff
@@ -0,0 +1,59 @@
+import { MetadataLink, MetadataLinkValid } from '@geonetwork-ui/util/shared'
+import { WfsEndpoint } from '@camptocamp/ogc-client'
+
+export const downloadFormats = {
+  csv: ['csv'],
+  geojson: ['geojson'],
+  json: ['json'],
+  shp: ['shp', 'shape'],
+  kml: ['kml'],
+  gpkg: ['gpkg', 'geopackage'],
+  excel: ['xls', 'xlsx'],
+  pdf: ['pdf'],
+  zip: ['zip'],
+}
+
+export function getFormat(link: MetadataLink, formats: any): string {
+  if ('format' in link) {
+    return link.format
+  }
+  for (const format in formats) {
+    for (const alias of formats[format]) {
+      if (findFileFormats(link, alias)) return format
+    }
+  }
+  return 'unknown'
+}
```

Comment:
Ok, I thought it could make sense to be broader here (eg. to be able to display the format of a shapefile instead of `zip`, which is most commonly its file extension). Do you fear we will really "missguess" formats with this approach?